### PR TITLE
Tighten break Cow tests and test helpers

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- assert the `Cow` variants returned by `format_breaks` so tests cover the borrowed-output optimisation directly
- update the break-formatting Rustdoc example to show borrowed thematic break output
- remove the redundant `string_vec!` helper and keep test helper lint suppressions item-scoped

## Testing
- `make check-fmt`
- `make lint`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_687837d741488322b8f005ad81122c27

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai pause

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:
## Individual Comments

### Comment 1
<location> `src/breaks.rs:31` </location>
<code_context>

         if !in_code && THEMATIC_BREAK_RE.is_match(line.trim_end()) {
-            out.push(THEMATIC_BREAK_LINE.clone());
+            out.push(Cow::Owned(THEMATIC_BREAK_LINE.clone()));
         } else {
-            out.push(line.clone());
</code_context>

<issue_to_address>
Cloning THEMATIC_BREAK_LINE for each thematic break may be inefficient.

Consider using Cow::Borrowed(THEMATIC_BREAK_LINE.as_str()) to avoid unnecessary allocations, since THEMATIC_BREAK_LINE is static.
</issue_to_address>
~~~

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (2 errors, 7 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Tests verify thematic break formatting but not the Cow optimisation. PartialEq bypasses variant checking, so tests pass regardless of actual Cow::Borrowed or Cow::Owned usage in the implementation. | Directly inspect Cow variant types in tests instead of relying on PartialEq comparison which ignores the allocation optimisation patterns the PR claims to implement. |
| Rust Compiler Lint Integrity | ❌ Error | tests/common/mod.rs retains `#![allow(unfulfilled_lint_expectations)]` on line 2, a crate-wide suppression violating the requirement to avoid broad lint suppressions. | Remove `#![allow(unfulfilled_lint_expectations)]` from line 2 of tests/common/mod.rs; existing #\[expect(...)\] attributes on items remain appropriate. |
| Title check | ⚠️ Warning | The PR title 'Refactor format\_breaks to use Cow' does not align with the actual changes, which focus on adding a string\_vec! macro and updating documentation examples. | Update the title to reflect the actual changeset: 'Add string\_vec! macro and update documentation examples' or similar to accurately represent the modifications made. |
| Description check | ⚠️ Warning | The PR description discusses refactoring format\_breaks to use Cow, which does not match the documented changes to documentation and test macros. | Align the description with the actual changes: adding string\_vec! macro, updating documentation examples, and simplifying test data creation. |
| User-Facing Documentation | ⚠️ Warning | Breaking API change to public `format_breaks()` return type (Vec → Vec) not documented in CHANGELOG or docs/users-guide.md. | Update CHANGELOG.md with the API change and add a migration guide to docs/users-guide.md explaining the new Cow return type for library users. |
| Developer Documentation | ⚠️ Warning | PR changes public API `format_breaks` to return `Vec<Cow<'_, str>>` for memory optimisation, but this architectural change lacks documentation in developers-guide or ADR. | Document breaks module in developers-guide explaining `Cow<str>` optimisation, or create ADR recording the memory allocation design decision. |
| Testing (Property / Proof) | ⚠️ Warning | The PR introduces an invariant for correct output whether using borrowed or owned Cow types, lacking property tests for varying inputs. | Add proptest to verify: output length equals input, non-thematic-break lines unchanged, thematic breaks always THEMATIC\_BREAK\_LEN. |
| Testing (Compile-Time / Ui) | ⚠️ Warning | New `string_vec!` macro added without any compile-time (trybuild) or doctest validation; no snapshot or unit tests verify macro behaviour. | Add doctest examples to the `string_vec!` macro demonstrating proper usage, or create unit tests validating the macro's compile-time expansion and runtime Vec construction. |
| Architectural Complexity And Maintainability | ⚠️ Warning | Introduces unused `string_vec!` macro duplicating `lines_vec!` (283 uses). Adds conceptual burden without reducing complexity or enabling reuse, violating the abstraction justification principle. | Remove the `string_vec!` macro definition and update the documentation example to use the established `lines_vec!` macro instead. |

**Comment by leynos** [comment]
@coderabbitai review

File: tests/prelude/mod.rs [hunk 1]
```diff
@@ -1,15 +1,14 @@
 //! Common imports for integration tests.
-#![allow(unfulfilled_lint_expectations)]
 
-#[expect(unused_imports, reason = "re-exporting common test utilities")]
+#[allow(unused_imports, reason = "re-exporting common test utilities")]
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/prelude/mod.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 6 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Tests for format\_breaks substantively verify Cow::Borrowed vs Owned distinction, but PR violates coding guidelines with 11 #\[allow\] attributes instead of #\[expect\]. | Replace all #\[allow(...)\] with #\[expect(...)\] in tests/common/mod.rs (7 instances) and tests/prelude/mod.rs (4 instances) per coding guidelines. |
| Unit Architecture | ❌ Error | Format\_breaks changes are architecturally sound, but 11 `#[allow]` attributes reduce visibility of compiler concerns where `#[expect]` would increase visibility per the custom check principle. | Replace all `#[allow]` with `#[expect]` attributes in tests/common/mod.rs and tests/prelude/mod.rs to make lint suppressions explicit per architectural visibility requirements. |
| Rust Compiler Lint Integrity | ❌ Error | PR changed 10 `#[expect(...)]` to `#[allow(...)]` lint attributes in tests/common/mod.rs and tests/prelude/mod.rs, violating the requirement that `#[allow]` is forbidden. | Replace all `#[allow(...)]` lint attributes with `#[expect(...)]` equivalents to preserve compiler detection of unused code. |
| User-Facing Documentation | ⚠️ Warning | Breaking change to format\_breaks return type (Vec → Vec) not documented in CHANGELOG.md or docs/users-guide.md; required by custom check for user-facing functionality changes. | Add CHANGELOG entry documenting format\_breaks return type change; update docs/users-guide.md to explain Cow optimisation and guide users on handling the new type. |
| Developer Documentation | ⚠️ Warning | The `format_breaks` return type changed from `Vec<String>` to `Vec<Cow<'_, str>>` but lacks documentation in `docs/developers-guide.md` and CHANGELOG.md. | Document the Cow allocation strategy in `docs/developers-guide.md` and add a CHANGELOG entry noting the `format_breaks` return type change. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Tests verify Cow borrowing correctly but lack edge case coverage (empty/single-line input, exactly 3-char thematic breaks, consecutive breaks, complex fence scenarios, output length invariants). | Add tests for edge cases: empty input, single lines, minimum-length thematic breaks (3 chars), consecutive breaks, nested fences, and property tests verifying output length equals input length and Cow::Borrowed for all unmodified lines. |
| Testing (Property / Proof) | ⚠️ Warning | PR introduces a Cow<'\_, str> allocation invariant spanning ranges of inputs and states, but contains zero property tests; proptest not added despite the invariant. | Add proptest verifying Cow::Borrowed for non-thematic lines, Cow::Borrowed(THEMATIC\_BREAK\_LINE.as\_str()) for breaks, and correct fence state transitions across arbitrary Markdown inputs. |
| Observability | ⚠️ Warning | The PR optimises format\_breaks to return Cow::Borrowed instead of Cow::Owned, an observable behaviour change affecting a public API, but provides no logging, metrics, or diagnostic information. | Document the Cow borrowing optimisation strategy to help callers understand variant handling; provide guidance on memory allocation implications for maintainers monitoring this function in production. |
| Architectural Complexity And Maintainability | ⚠️ Warning | PR introduces justified test macros that reduce complexity, but violates coding guidelines by replacing #\[expect\] with #\[allow\] lint attributes in tests/common/mod.rs and tests/prelude/mod.rs. | Replace all #\[allow\] lint attributes with #\[expect(lint, reason = "...")\] across tests/common/mod.rs (7 instances) and tests/prelude/mod.rs (4 instances) per repository coding guidelines. |

File: tests/cli.rs [hunk 2]
```diff
@@ -11,12 +11,18 @@ use std::{
...
 
 #[macro_use]
 mod prelude;
-use prelude::*;
+
+#[rstest::fixture]
+fn broken_table() -> Vec<String> {
+    let lines = lines_vec!["| A | B |    |", "| 1 | 2 |  | 3 | 4 |"];
+    lines
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/cli.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/parallel.rs [hunk 3]
```diff
@@ -2,12 +2,21 @@
...
-use prelude::*;
+#[path = "support/cli_args.rs"]
+mod cli_args;
+use cli_args::run_cli_with_args;
+
+#[rstest::fixture]
+fn broken_table() -> Vec<String> {
+    let lines = lines_vec!["| A | B |    |", "| 1 | 2 |  | 3 | 4 |"];
+    lines
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/parallel.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/table/mod.rs [hunk 4]
```diff
@@ -17,10 +17,15 @@
...
+use rstest::fixture;
 
 #[macro_use]
 mod prelude;
-use prelude::*;
+
+#[fixture]
+fn broken_table() -> Vec<String> {
+    lines_vec!["| A | B |    |", "| 1 | 2 |  | 3 | 4 |"]
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/table/mod.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 5 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Module-Level Documentation | ❌ Error | The `prop_tests` module in src/breaks.rs (line 108) lacks a module-level doc comment, violating the requirement that all modules must carry documentation explaining their purpose and utility. | Add a `//!` module-level doc comment to the `prop_tests` module explaining it contains property-based tests for thematic break formatting and Cow allocation semantics. |
| Unit Architecture | ❌ Error | The `prop_tests` module in `src/breaks.rs` lacks required module-level `//!` doc comment. Every module must document its purpose and utility. | Add module-level doc comment to `mod prop_tests` explaining the property-based testing strategy and verification approach for borrowing semantics. |
| Rust Compiler Lint Integrity | ❌ Error | The `prop_tests` module in `src/breaks.rs` line 109 lacks required `//!` doc comment, violating module documentation guidelines and review feedback. | Add `//! Property-based tests for thematic break formatting.` doc comment to `prop_tests` module before use statements. |
| User-Facing Documentation | ⚠️ Warning | Breaking API change to `format_breaks` return type documented in CHANGELOG and rustdoc but not in user-facing README documentation for library users. | Add a note in README's "Library usage" section documenting the `format_breaks` return type change from `Vec<String>` to `Vec<Cow<'_, str>>` with guidance on calling `.into_owned()` if needed. |
| Developer Documentation | ⚠️ Warning | The `prop_tests` module in src/breaks.rs (line 108) lacks a required `//!` module-level doc comment per coding guidelines, though the Cow allocation strategy is documented in developers-guide.md. | Add a `//!` module-level doc comment to the `prop_tests` module explaining its purpose: "Property-based tests for thematic break formatting. Verifies `format_breaks` borrowing semantics and invariants using generated test cases." |
| Testing (Unit And Behavioural) | ⚠️ Warning | The `prop_tests` module in src/breaks.rs lacks a required module-level `//!` doc comment. Review feedback explicitly flagged this requirement. | Add `//!` module documentation to the `prop_tests` module explaining its purpose. The module-level doc comment should describe it as property-based tests for verifying `format_breaks` borrowing semantics. |
| Domain Architecture | ⚠️ Warning | Domain architecture is properly maintained with no vendor/infrastructure concerns in core code. However, `prop_tests` module (src/breaks.rs:108) lacks required module-level doc comment. | Add module-level `//!` documentation to the `prop_tests` module explaining its purpose, e.g. "Property-based tests for thematic break formatting." |
| Architectural Complexity And Maintainability | ⚠️ Warning | Four review-flagged architectural issues remain unaddressed: missing prop\_tests doc, incomplete broken\_table consolidation, unpinned proptest, and non-compliant spelling. | Add prop\_tests module doc comment; simplify broken\_table in cli.rs/parallel.rs; pin proptest to "1.11.0"; correct "Synthesised" to "Synthesized". |

**Comment by leynos** [comment]
@coderabbitai "Synthesized" is correct for en-GB-oxendict (-ize / -yse / -our)

**Comment by leynos** [comment]
@coderabbitai review

File: src/breaks.rs [hunk 5]
```diff
@@ -19,7 +19,7 @@ static THEMATIC_BREAK_LINE: std::sync::LazyLock<String> =
 /// Normalize thematic breaks outside fenced code blocks.
 ///
 /// Consecutive hyphens, asterisks or underscores are replaced with a
-/// standardised line of underscores. Fenced code blocks are ignored so
+/// standardized line of underscores. Fenced code blocks are ignored so
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/breaks.rs:22]: @coderabbitai "standardized" is correct for en-GB-oxendict (-ize / -yse / -our)

File: test-macros/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        #[cfg_attr(
+            clippy,
+            expect(
+                clippy::allow_attributes,
+                reason = "needed to allow unused_braces for fixture macro expansion"
+            )
+        )]
+        #parsed_item
+    }
+    .into()
```
[the remark this answers is not in the document]
Comment (leynos) [reply test-macros/src/lib.rs:31]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/cli_args.rs [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+/// Run the `mdtablefix` binary with the provided arguments.
+///
+/// Returns an [`Assert`] handle for chaining output and status checks.
+pub fn run_cli_with_args(args: &[&str]) -> Assert {
+    Command::cargo_bin("mdtablefix")
+        .expect("failed to create command")
+        .args(args)
+        .assert()
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/cli_args.rs:10]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/cli_stdin.rs [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+/// Run the `mdtablefix` binary with the provided arguments and standard input.
+///
+/// Returns an [`Assert`] handle for chaining output and status checks.
+pub fn run_cli_with_stdin(args: &[&str], input: &str) -> Assert {
+    Command::cargo_bin("mdtablefix")
+        .expect("failed to create command")
+        .args(args)
+        .write_stdin(input.to_owned())
+        .assert()
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/cli_stdin.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/cli_stdin.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 6 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Five unresolved review comments: toggle\_code\_span bug, "standardized"→"standardised" spelling, #\[allow\]→#\[expect\] in test-macros, and .expect() misuse in CLI helpers. | Fix toggle\_code\_span else-if logic; fix spelling; emit #\[expect\] not #\[allow\]; return Result from CLI helpers instead of .expect(). |
| Unit Architecture | ❌ Error | CLI helpers hide fallibility via `.expect()` rather than returning `Result`. Proc-macro uses forbidden `#[allow]`. Backtick span logic incorrectly toggles nested backticks. | Make CLI helpers return `Result`. Use `#[expect]` not `#[allow]`. Fix `toggle_code_span` to `else if open.is_none()`. Change "standardized" to "standardised". |
| Rust Compiler Lint Integrity | ❌ Error | Violations: test-macros uses `#[allow(...)]` instead of `#[expect(...)]`; cli helpers use `.expect()`; toggle\_code\_span lacks guard for nested backticks; "standardized" should be "standardised". | Use `#[expect(...)]` not `#[allow(...)]` in test-macros. Return `Result` from cli helpers. Guard `toggle_code_span` with `else if open.is_none()`. Change spelling to "standardised". |
| Developer Documentation | ⚠️ Warning | New proc-macro crate (test-macros) and test support reorganisation lack documentation in developers-guide; architectural decisions for test-tooling changes are not recorded. | Document test-macros purpose/usage and test-support reorganisation in docs/developers-guide.md; consider ADR if test-macros warrants architectural significance. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Tests appropriately verify Cow borrowing and edge cases. However, multiple review comments remain unresolved: toggle\_code\_span logic, standardised spelling, and error handling in test helpers. | Fix toggle\_code\_span else-if logic, change standardized→standardised, replace #\[allow\] with #\[expect\], and return Result from cli helpers instead of .expect(). |
| Testing (Property / Proof) | ⚠️ Warning | Property tests cover borrowing invariants but lack coverage for FenceTracker combined with Cow semantics in mixed fence/break scenarios. | Add property test with composite strategy generating fence-marker/break/fence-marker sequences, asserting Cow::Borrowed semantics across fence state transitions. |
| Testing (Compile-Time / Ui) | ⚠️ Warning | New proc-macro `allow_fixture_expansion_lints` lacks trybuild compile-time tests. Repo has established trybuild infrastructure; tests/ui/ must include test cases. | Add tests/ui/allow\_fixture\_expansion\_lints\_pass.rs to verify proc macro expansion and attribute emission. Include trybuild test case in tests/compile.rs calling cases.pass(...). |
| Performance And Resource Use | ⚠️ Warning | Test helpers have allocation regressions: toggle\_code\_span logic error, scan\_code\_spans Vec allocation per line, cli\_stdin unnecessary to\_owned(). | Fix toggle\_code\_span else branch, replace Vec with enumerate(), remove to\_owned() in cli\_stdin. |
| Architectural Complexity And Maintainability | ⚠️ Warning | test-macros proc-macro crate used only once for single fixture; unnecessary abstraction over direct #\[expect\] attribute. Review comments unaddressed. | Remove test-macros crate; use #\[expect(unused\_braces, reason=...)\] on broken\_table. Fix toggle\_code\_span else/else-if bug, standardised spelling, #\[allow\]→#\[expect\], and .expect() in cli helpers. |

**Comment by leynos** [comment]
@coderabbitai this is required, due to a bug in `rstest`. Due to the nature of macro expansion, `#[allow]` is unavoidable. Trust me, I have tried.

> ## 1  test-macros: emit `#[expect]` instead of `#[allow]`

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 5 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Five unresolved review comments: toggle\_code\_span bug, "standardized"→"standardised" spelling and .expect() misuse in CLI helpers. | Fix toggle\_code\_span else-if logic; fix spelling; return Result from CLI helpers instead of .expect(). |
| Unit Architecture | ❌ Error | CLI helpers hide fallibility via `.expect()` rather than returning `Result`. Backtick span logic incorrectly toggles nested backticks. | Make CLI helpers return `Result`. Fix `toggle_code_span` to `else if open.is_none()`. Change "standardized" to "standardised". |
| Rust Compiler Lint Integrity | ❌ Error | Violations: cli helpers use `.expect()`; toggle\_code\_span lacks guard for nested backticks; "standardized" should be "standardised". | Return `Result` from cli helpers. Guard `toggle_code_span` with `else if open.is_none()`. Change spelling to "standardised". |
| Developer Documentation | ⚠️ Warning | New proc-macro crate (test-macros) and test support reorganisation lack documentation in developers-guide; architectural decisions for test-tooling changes are not recorded. | Document test-macros purpose/usage and test-support reorganisation in docs/developers-guide.md, noting that this is 1) required due to a bug in rstest's `#[fixture]` macro (tracking ticket forthcoming) that leads to a conflict between the single line function formatter and the unused braces lint, and 2) the emitted `#[allow]` is required rather than `#[expect]` because of the way Rust procmacros function. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Tests appropriately verify Cow borrowing and edge cases. However, multiple review comments remain unresolved: toggle\_code\_span logic and standardised spelling. | Fix toggle\_code\_span else-if logic, change standardized→standardised, and return Result from cli helpers instead of .expect(). |
| Testing (Property / Proof) | ⚠️ Warning | Property tests cover borrowing invariants but lack coverage for FenceTracker combined with Cow semantics in mixed fence/break scenarios. | Add property test with composite strategy generating fence-marker/break/fence-marker sequences, asserting Cow::Borrowed semantics across fence state transitions. |
| Testing (Compile-Time / Ui) | ⚠️ Warning | New proc-macro `allow_fixture_expansion_lints` lacks trybuild compile-time tests. Repo has established trybuild infrastructure; tests/ui/ must include test cases. | Add tests/ui/allow\_fixture\_expansion\_lints\_pass.rs to verify proc macro expansion and attribute emission. Include trybuild test case in tests/compile.rs calling cases.pass(...). |
| Performance And Resource Use | ⚠️ Warning | Test helpers have allocation regressions: toggle\_code\_span logic error, scan\_code\_spans Vec allocation per line, cli\_stdin unnecessary to\_owned(). | Fix toggle\_code\_span else branch, replace Vec with enumerate(), remove to\_owned() in cli\_stdin. |

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 7 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Core Cow tests are rigorous with pointer equality checks; test infrastructure has two defects: wrong attribute order in fixtures.rs and missing validation in assert\_wrapped\_blockquote. | Swap attribute order in tests/support/fixtures.rs: #\[test\_macros::allow\_fixture\_expansion\_lints\] above #\[rstest::fixture\]. Add expected > 0 guard to assert\_wrapped\_blockquote with updated documentation. |
| Unit Architecture | ❌ Error | Fixture attribute order reversed and assert\_wrapped\_blockquote missing expected > 0 validation plus incomplete docs versus assert\_wrapped\_list\_item. | Swap fixture attributes; add expected > 0 check to assert\_wrapped\_blockquote and match documentation of assert\_wrapped\_list\_item. |
| Rust Compiler Lint Integrity | ❌ Error | Unresolved review comments: attribute order reversed in `tests/support/fixtures.rs`; missing `expected > 0` validation in `assert_wrapped_blockquote`. | Swap proc-macro attributes in `tests/support/fixtures.rs`. Add `expected > 0` check to `assert_wrapped_blockquote` with updated documentation. |
| User-Facing Documentation | ⚠️ Warning | Breaking change to public API function `format_breaks` documented in README.md and developers-guide.md, but not in docs/users-guide.md as explicitly required by the check. | Document the `format_breaks` breaking change in docs/users-guide.md or add a Library API section addressing the return type change for library consumers. |
| Developer Documentation | ⚠️ Warning | Internal test API change not documented: `lines_vec!` and `include_lines!` macros are now `#[macro_export]` but lack architectural documentation in `docs/developers-guide.md`. | Document the exported macros in Test infrastructure section, explaining their purpose and why they are exported for test utilities. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Two review comments unaddressed: fixture attributes are reversed; assert\_wrapped\_blockquote lacks expected>0 validation guard. | Swap attribute order in fixtures.rs. Add expected>0 check to assert\_wrapped\_blockquote and update documentation. |
| Testing (Compile-Time / Ui) | ⚠️ Warning | Trybuild tests present, but two unresolved prior review issues remain: fixture attributes in wrong order and assert\_wrapped\_blockquote lacks expected parameter validation. | Reorder attributes in tests/support/fixtures.rs (allow\_fixture\_expansion\_lints above rstest::fixture). Add expected > 0 guard and update docs in assert\_wrapped\_blockquote. |
| Performance And Resource Use | ⚠️ Warning | Two unaddressed review comments: fixture attributes in wrong order and assert\_wrapped\_blockquote lacks validation and documentation updates. | Flip attribute order in tests/support/fixtures.rs to list allow\_fixture\_expansion\_lints first. Add expected > 0 validation to assert\_wrapped\_blockquote and update its doc comment. |
| Concurrency And State | ⚠️ Warning | PR introduces shared static THEMATIC\_BREAK\_LINE accessed concurrently via rayon, but lacks explicit concurrent tests exercising LazyLock initialisation races. | Add thread-based tests spawning multiple threads accessing format\_breaks simultaneously, verifying LazyLock race-safety and pointer stability under concurrent access. |
| Architectural Complexity And Maintainability | ⚠️ Warning | Unresolved review: fixtures.rs attribute order wrong; allow\_fixture\_expansion\_lints must wrap rstest::fixture. wrap\_assertions.rs missing expected > 0 guard. | Swap attributes in fixtures.rs. Add expected > 0 check and docs to assert\_wrapped\_blockquote. |

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### src/breaks.rs (1)

`22-22`: _⚠️ Potential issue_ | _🟡 Minor_ | _⚡ Quick win_
**Correct the spelling to match en-GB-oxendict guidelines.**
Line 22 uses "standardised" but the coding guidelines mandate en-GB-oxendict spelling, which uses `-ize` endings. Change to "standardized".
As per coding guidelines: "Comments and docs must follow en-GB-oxendict (-ize / -yse / -our) spelling and grammar (examples: 'recognize,' 'synthesized,' 'polymerization,' 'analyse,' 'colour,' 'behaviour')."

##### 🔧 Proposed fix

```diff
-/// Consecutive hyphens, asterisks or underscores are replaced with a
-/// standardised line of underscores. Fenced code blocks are ignored so
+/// Consecutive hyphens, asterisks or underscores are replaced with a
+/// standardized line of underscores. Fenced code blocks are ignored so
```

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@src/breaks.rs` at line 22, Change the comment text in breaks.rs that
currently uses the en-GB “standardised” spelling to the en-GB-oxendict-compliant
“standardized”; locate the comment line containing "standardised line of
underscores. Fenced code blocks are ignored so" and replace "standardised" with
"standardized" to conform to the required -ize spelling in documentation
comments.

**Comment by leynos** [comment]
@coderabbitai review
